### DISCUSSION TITLE
`mongoose.connection.model` return instanceof `Model`

### DISCUSSION THREAD

**Discussion by hasezoey** [Automattic/mongoose#12179] [Q&A] [answered]
I could not find a way to test the return from `mongoose.connection.model` against any `Model` "class" i could find, it would always return `false`, so i wanted to ask if there is a way to match this model type against any model class? (because `Connection.prototype.Model` does not exist)

example:
```ts
const model = mongoose.connection.model('test', new mongoose.Schema());

console.log(model instanceof mongoose.Model); // false
console.log(model instanceof mongoose.connection.Model); // this property does not exist
console.log(model instanceof mongoose.connection.base.Model); // false
```

**Comment by vkarpov15** [comment, marked as answer]
`mongoose.connection.model` returns a "class" that extends from `mongoose.Model`, **not** an instance of `mongoose.Model`. `console.log(Object.getPrototypeOf(model) === mongoose.Model); // true`

**Reply by hasezoey** [reply]
> "class" that extends

with the normal interpretation of that, shouldnt it then still be a instance of the model class if it just "extends" it?

like normal js
```js
class A {}
class B extends A {}

console.log(new B() instanceof A); // true
```

**Reply by vkarpov15** [reply]
Yeah, if you do `const TestModel = mongoose.connection.model('Test', schema); const doc = new TestModel(); doc instanceof mongoose.Model; // true`

**Reply by hasezoey** [reply]
thanks, forgot again that a `document` is a instance of `Model` and not `model`(compiled from schema) a instance of `Model`(statically provided by mongoose)
or said in different terms, i forgot that the models returned by `.model` are *extensions* of `Model` and documents are *instances* of the models
